Functions/xASL_test_Flavors.m

```diff
@@ -29,179 +30,84 @@ function xASL_test_Flavors(pathExploreASL, pathTest, bTest, x)
+    if bTest(2)
+        flavors.loggingTable = xASL_test_Flavors_DCM2BIDS(testConfig, x, flavors.loggingTable);
+    end
     
-    if xASL_exist(flavorsPath,'dir')
-        fList = xASL_adm_GetFileList(flavorsPath, [], 'List', [], 1);
-    else
-        error('No FlavorDatabase folder found...');
+
+    %% 3. Run the comparison of converted BIDS with the reference data
+    if bTest(3)
+        fprintf('\n================================== CHECK THE BIDS CONVERSION =================================\n');
+        [flavors,~] = xASL_test_Flavors_Compare(testConfig,flavors,'rawdata','rawdataReference');
     end
-    fprintf('  '); % Make empty spaces for xASL_TrackProgress
-	for iList = 1:length(fList)
-        % Display the overall progress and create the conversion directories
-		xASL_TrackProgress(iList, length(fList));
-		xASL_adm_CreateDir(fullfile(conversionPath,fList{iList}));
-		xASL_adm_CreateDir(fullfile(conversionPath,fList{iList}, 'sourcedata'));
-		
-        % Copy the sourcedata to the temporary conversion folder
-		SourceDir = fullfile(flavorsPath, fList{iList}, 'sourcedata');
-        DestinationDir = fullfile(conversionPath, fList{iList}, 'sourcedata');
-        xASL_Copy(SourceDir, DestinationDir, 1); 
-		
-        % Copy the JSON meta files to the temporary conversion folder
-		xASL_Copy(fullfile(flavorsPath,fList{iList},'sourceStructure.json'), fullfile(conversionPath,fList{iList},'sourceStructure.json'), 1);
-		xASL_Copy(fullfile(flavorsPath,fList{iList},'studyPar.json'), fullfile(conversionPath,fList{iList},'studyPar.json'), 1);
-		if exist(fullfile(flavorsPath,fList{iList},'dataPar.json'),'file')
-			xASL_Copy(fullfile(flavorsPath,fList{iList},'dataPar.json'), fullfile(conversionPath,fList{iList},'dataPar.json'), 1);
-		end
-	end
-	fprintf('\n');
-	
-	% Create a copy of the reference BIDS data
-	fprintf('Copying the BIDS reference data to a temporary folder: %s\n', referencePath);
-	xASL_adm_CreateDir(referencePath);
-	
-	fList = xASL_adm_GetFileList(flavorsPath, [], 'List', [], 1);
-    fprintf('  '); % Make empty spaces for xASL_TrackProgress
-	for iList = 1:length(fList)
-		xASL_TrackProgress(iList,length(fList));
-		xASL_adm_CreateDir(fullfile(referencePath,fList{iList}));
-		xASL_adm_CreateDir(fullfile(referencePath,fList{iList}, 'rawdata'));
+    
 
-        % Copy rawdata folder for comparison
-        xASL_Copy(fullfile(flavorsPath,fList{iList}, 'rawdata'), fullfile(referencePath,fList{iList}, 'rawdata'), 1); 
+    %% 4. Run the BIDS to Legacy conversion
+    if bTest(4)
+        xASL_test_Flavors_BIDS2LEGACY(testConfig);
     end
-    fprintf('\n');
-end
+    
 
-%% 2. Run the conversion of source data to BIDS
-if bTest(2)
-	xASL_test_Flavors_DCM2BIDS(conversionPath, x);
-end
+    %% 5. Run the comparison of data converted to the legacy format with the reference data
+    if bTest(5)
+        fprintf('\n================================= CHECK THE LEGACY CONVERSION ================================\n');
```

Comment:
Complains about missing 
Warning: Missing file /ExploreASL/import_summary.csv  
We should add these CSV to exceptions - at least for checking BIDS2Legacy conversion

Reply:
This one as well|
Warning: Missing file /ExploreASL/bidsReportASL.json

Reply:
And the final logContent doesn't say anything about these - should we or should we not add the comparison differences to the final LogContent?

Reply:
Hey @jan-petr, yep saw those before. I was thinking that we could rather remove those from the flavors.comparisonTable instead. I thought that's not super high priority though, since we basically know about these things.
My wish would be to get rid of the logContent table and to mostly use flavors.comparisonTable and flavors.loggingTable instead. The overview there is a lot nicer and it works a in a more robust way. All of this log file scanning using regular expressions of XML and HTML text is super unstable.

Reply:
@jan-petr: check out this commit https://github.com/ExploreASL/ExploreASL/pull/835/commits/fd2fa5537078105a54d79dbf2ae5c4ee5355e142